test/extended/machines/machines.go

```diff
@@ -100,6 +116,76 @@ var _ = g.Describe("[sig-cluster-lifecycle][Feature:Machines] Managed cluster sh
 			e2e.Failf("Machine resources missing for nodes: %s", strings.Join(nodeNames.List(), ", "))
 		}
 	})
+
+	g.It("[sig-scheduling][Early] control plane machine set operator should not cause an early rollout", func() {
+		machineClientSet, err := machineclient.NewForConfig(oc.KubeFramework().ClientConfig())
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		pattern := `^([a-zA-Z0-9]+-)+master-\d+$`
+
+		g.By("checking for the openshift machine api operator")
+		skipUnlessMachineAPIOperator(dc, c.CoreV1().Namespaces())
+
+		g.By("ensuring every node is linked to a machine api resource")
+		allControlPlaneMachines, err := machineClientSet.MachineV1beta1().Machines("openshift-machine-api").List(context.Background(), metav1.ListOptions{
+			LabelSelector: masterMachineLabelSelector,
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		regex, err := regexp.Compile(pattern)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for _, m := range allControlPlaneMachines.Items {
+			matched := regex.MatchString(m.Name)
+			o.Expect(matched).To(o.BeTrue(), fmt.Sprintf("unexpected name of a control machine occured during early stages: %s", m.Name))
+		}
+	})
+
+	g.It("[sig-scheduling][Early] control plane machine set operator should not have any events", func() {
+		ctx := context.Background()
+
+		configClient, err := configclient.NewForConfig(oc.KubeFramework().ClientConfig())
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		infrastructure, err := configClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		platform := infrastructure.Status.PlatformStatus.Type
+
+		switch platform {
+		case v1.AWSPlatformType,
+			v1.AzurePlatformType,
+			v1.GCPPlatformType,
+			v1.NutanixPlatformType,
+			v1.OpenStackPlatformType:
+			// Continue with the test
+		default:
+			g.Skip(fmt.Sprintf("Skipping test on platform: %s", platform))
+		}
+
+		machineClientSet, err := machineclient.NewForConfig(oc.KubeFramework().ClientConfig())
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("checking for the openshift machine api operator")
+		skipUnlessMachineAPIOperator(dc, c.CoreV1().Namespaces())
+
+		g.By("getting the control plane machine set")
+		cpmsClient := machineClientSet.MachineV1().ControlPlaneMachineSets("openshift-machine-api")
+		cpms, err := cpmsClient.Get(ctx, "cluster", metav1.GetOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("getting the events from the control plane machine set")
+		eventsClient := oc.KubeClient().CoreV1().Events("openshift-machine-api")
+		cpmsEvents, err := eventsClient.List(ctx, metav1.ListOptions{
+			TypeMeta: metav1.TypeMeta{
+				Kind: cpms.Kind,
+			},
+		})
+		o.Expect(err).ToNot(o.HaveOccurred())
```

Comment:
Filtering based on TypeMeta doesn't work, you'll need to filter on the involved object once you've done the listing of the events

Reply:
Yes, changed it to filter based on fieldSelector and the objectInvolved.Kind/Name.